server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -816,9 +844,11 @@ public DiscoveryNodes build() {
                 filteredNodes(nodes, DiscoveryNode::isIngestNode),
                 masterNodeId,
                 localNodeId,
-                minNonClientNodeVersion == null ? Version.CURRENT : minNonClientNodeVersion,
-                maxNodeVersion == null ? Version.CURRENT : maxNodeVersion,
-                minNodeVersion == null ? Version.CURRENT.minimumCompatibilityVersion() : minNodeVersion
+                Objects.requireNonNullElse(minNonClientNodeVersion, Version.CURRENT),
```

Comment:
TIL requireNoneNullElse, didn't know there was an "else" version, nice!

Reply:
I don't like the name myself, but its very useful